worldengine/simulations/temperature.py

```diff
@@ -89,13 +92,8 @@ def _calculate(world, seed, elevation, mountain_level):
                                    base=base) * (border - x) / border)
 
                 t = (latitude_factor * 12 + n * 1) / 13.0 / distance_to_sun
-                if elevation[y, x] > mountain_level:  # vary temperature based on height
-                    if elevation[y, x] > (mountain_level + 29):
-                        altitude_factor = 0.033
-                    else:
-                        altitude_factor = 1.00 - (
-                            float(elevation[y, x] - mountain_level) / 30)
-                    t *= altitude_factor
-                temp[y, x] = t
+
+                # vary temperature based on height
+                temp[y, x] = t * numpy.interp(e[y, x], [mountain_level, highest_mountain_level], [1.00, min_altitude_factor])
```

Comment:
Does basically the same as the old code, but there are not arbitrary constants used; instead only heights are taken into account. `min_altitude_factor` could later be made global.